sdk/objc/components/audio/RTCAudioSession+Configuration.mm

```diff
@@ -133,7 +136,9 @@ - (BOOL)setConfiguration:(RTC_OBJC_TYPE(RTCAudioSessionConfiguration) *)configur
 
   if (self.isActive &&
       // TODO(tkchin): Figure out which category/mode numChannels is valid for.
-      [self.mode isEqualToString:AVAudioSessionModeVoiceChat]) {
+      // RingRTC change to support VoiceChat and VideoChat in objc sdk.
+      ([self.mode isEqualToString:AVAudioSessionModeVoiceChat] ||
+       [self.mode isEqualToString:AVAudioSessionModeVideoChat])) {
```

Comment:
Should we change the iOS client to also set the number of channels?  Otherwise, wouldn't we run into a similar problem in the future if the iOS clients decides to and then it conflicts?  It just seems better to have either one side control everything or the other.   And if iOS is going to control most of them, why not all?

Reply:
This part has to be done after the session is made `active`. I think it would be quite a bit more effort to pull those parts out.

The audio controller that manages the device is part of the SDK here, and does some other things to adjust the sample rate, (maybe) channels, and IO Buffer settings and handle different issues that come up. The client could do some of these things too, but it seems like additional risk at this point. Maybe plan for a possible future refactor, when the client gets some of its own consolidation?